src/agent.rs

```diff
@@ -106,32 +106,35 @@ impl Agent {
     ///     .call();
     /// println!("{:?}", r);
     /// ```
-    pub fn request(&self, method: &str, path: &str) -> Request {
-        Request::new(self.clone(), method.into(), path.into())
+    pub fn request(&self, method: &str, url: &str) -> RequestBuilder {
+        RequestBuilder::new()
+            .agent(self.clone())
+            .method(method)
+            .url_str(url)
```

Comment:
The `RequestBuilder` I'm not so sure about, the reasoning being: who are we doing this for?

1. From the user's perspective, I'd like them to interact with two main ureq objects, `Request` and `Response`. However with the `RequestBuilder`, it's like `Request` becomes an almost invisible intermediate object that the user only would interact with if they for some reason `.build()` instead of call it directly. 
2. Optics of the API. `pub fn get(path: &str) -> Request` conveys the simplicity I want ureq to have. I like there is no `Result` and no intermediate object that becomes a mental hurdle to grasp (granted, builders are a common pattern).
3. The Request was already a builder pattern. Not in name, but I feel spelling that out is more a formalism for our benefit than the user's.

What are the arguments for a `RequestBuilder`?